test/mobile/test_bytecode.py

```diff
@@ -15,33 +25,260 @@
+            backport_success = _backport_for_mobile(output_model_path, output_model_path_backport, 4)
+            assert(backport_success)
+
+            backport_version = _get_model_bytecode_version(output_model_path_backport)
+            assert(backport_version == 4)
+
+            shutil.rmtree(tmpdirname)
+
+
+    def test_backport_bytecode_from_file_to_file(self):
+        script_module_v5_path = pytorch_test_dri / "cpp" / "jit" / "script_module_v5.ptl"
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            backport_model_path = Path(tmpdirname, "backport_script_module_v5.ptl")
+            # backport from file
+            success = _backport_for_mobile(script_module_v5_path, backport_model_path, 4)
+            assert(success)
+
+            buf = io.StringIO()
+            torch.utils.show_pickle.main(["", tmpdirname + "/" + backport_model_path.name + "@*/bytecode.pkl"], output_stream=buf)
+            output = buf.getvalue()
+
+            expected_result = SCRIPT_MODULE_V4_path_BYTECODE_PKL
+            acutal_result_clean = "".join(output.split())
+            expect_result_clean = "".join(expected_result.split())
+            isMatch = fnmatch.fnmatch(acutal_result_clean, expect_result_clean)
+            assert(isMatch)
+
+            # Load model v5 and run forward method
+            mobile_module = _load_for_lite_interpreter(str(backport_model_path))
+            module_input = 1
+            mobile_module_result = mobile_module(module_input)
+            expected_mobile_module_result = 3 * torch.ones([2, 4], dtype=torch.float64)
+            torch.testing.assert_allclose(mobile_module_result, expected_mobile_module_result)
+            shutil.rmtree(tmpdirname)
+
+
+    def test_backport_bytecode_from_file_to_buffer(self):
+        script_module_v5_path = pytorch_test_dri / "cpp" / "jit" / "script_module_v5.ptl"
+
+        # Backport model to v4
+        script_module_v4_buffer = _backport_for_mobile_to_buffer(script_module_v5_path, 4)
+        buf = io.StringIO()
+
+        # Check version of the model v4 from backport
+        bytesio = io.BytesIO(script_module_v4_buffer)
+        backport_version = _get_model_bytecode_version(bytesio)
+        assert(backport_version == 4)
 
-        # Load model v4 and v5 and run forward method
-        jit_module_v4 = torch.jit.load(str(script_module_v4))
-        mobile_module_v4 = _load_for_lite_interpreter(str(script_module_v4))
+        # Load model v4 from backport and run forward method
+        bytesio = io.BytesIO(script_module_v4_buffer)
+        mobile_module = _load_for_lite_interpreter(bytesio)
+        module_input = 1
+        mobile_module_result = mobile_module(module_input)
+        expected_mobile_module_result = 3 * torch.ones([2, 4], dtype=torch.float64)
+        torch.testing.assert_allclose(mobile_module_result, expected_mobile_module_result)
+
+
+    def test_backport_bytecode_from_buffer_to_buffer(self):
```

Comment:
Do we have to check all combinations of file buffer? I am just wondering if this may end up becoming slow tests and fail on infra errors not giving good signal.

Reply:
Let's just run this for one model + one backport operation, simply to test the api functionality.